### DISCUSSION TITLE
Deleting data in bookkeeper

### DISCUSSION THREAD

**Discussion by ub-3** [apache/pulsar#23032] [Q&A] [answered]
1. Will the ledger be deleted if a rollover occurred according to the parameters: managedLedgerMaxLedgerRolloverTimeMinutes, m
anagedLedgerMinLedgerRolloverTimeMinutes, managedLedgerMaxEntriesPerLedger, but the cursor ledger was not closed.
2. And another question. If compaction is disabled on the bookkeeper side, will the entry log be deleted when closing ledger?

**Comment by sbourkeostk** [comment, marked as answer]
1. A topic (managed-ledger) consists of a set of ledgers. When a ledger is rolled over the previous ledger is only removed if all cursors have moved past its entries (assuming no retention/TTLs).
2. BookKeeper has no concept/knowledge of compaction, that's a Pulsar concept. As in question 1, a ledger is only deleted when the broker that responsible for the topic deletes the ledger from bookkeeper. The broker will not do that until: all messages stored in the ledger have been ack'd on all subscriptions or the TTL has expired, and the retention window has passed.
